cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueAdmin.java

```diff
@@ -192,21 +193,20 @@ public void truncate(String name) throws Exception {
     deleteConsumerConfigurations(queueName);
   }
 
-  private void truncate(byte[] tableNameBytes) throws IOException {
+  private void truncate(TableId tableId) throws IOException {
     HBaseAdmin admin = getHBaseAdmin();
-    if (admin.tableExists(tableNameBytes)) {
-      HTableDescriptor tableDescriptor = admin.getTableDescriptor(tableNameBytes);
-      admin.disableTable(tableNameBytes);
-      admin.deleteTable(tableNameBytes);
-      admin.createTable(tableDescriptor);
+    if (tableUtil.tableExists(admin, tableId)) {
+      HTableDescriptor tableDescriptor = tableUtil.getHTableDescriptor(admin, tableId);
```

Comment:
I got confused by the naming since you have to method named the same, but actually one is create, one is get. You should name them as:

```java
getHTableDescriptor(HBaseAdmin, TableId tableId);   // Get existing table descriptor
createHTableDescriptor(TableId tableId);                     // Create new instance of HTableDescriptor for the given tableId
```